### PR TITLE
make permitted_attributes support action

### LINKED ISSUE #2352: the permitted_attributes may need pass current action
https://github.com/thoughtbot/administrate/blob/062ea9f517b140ea04e2c6c4ff3a00a563bafcac/lib/generators/administrate/dashboard/templates/controller.rb.erb#L39

the permitted_attributes may need pass current action to support "FORM_ATTRIBUTES_NEW" and "FORM_ATTRIBUTES_EDIT". 

otherwise only go through "FORM_ATTRIBUTES" whatever edit or new action.

### PR DESCRIPTION
**Pull request by jubilee2**
fix #2352 and #2066

### CODE REVIEW

File: lib/administrate/base_dashboard.rb [hunk 1]
```diff
@@ -62,11 +62,18 @@ def specific_form_attributes_for(action)
...
 
-    def permitted_attributes
-      form_attributes.map do |attr|
+    def permitted_attributes(action = nil)
+      action =
+        case action
+        when "update" then "edit"
+        when "create" then "new"
+        else action
+        end
```
Comment (pablobm) [inline lib/administrate/base_dashboard.rb:71]: This code also exists in `lib/administrate/page/form.rb` where it's also used ahead of calling `form_attributes`. What do you think of removing it from both places and putting it at the start of `form_attributes` instead?

**Review by pablobm — COMMENTED** [review]
Looks good, thank you! I have left one comment and I think this should be good to merge soon :+1:

File: lib/administrate/base_dashboard.rb (hunk 1 above)
Comment (jubilee2) [reply lib/administrate/base_dashboard.rb]: done

File: spec/controllers/admin/customers_controller_spec.rb [hunk 2]
```diff
@@ -66,6 +66,11 @@ def default_sorting_direction
         expect(locals[:resources].map(&:name)).to eq sorted_customer_names
       end
     end
+
+    it "assigns the action name as a string" do
+      get :index
+      expect(controller.action_name).to be_a(String)
+    end
```
Comment (pablobm) [inline spec/controllers/admin/customers_controller_spec.rb:73]: Can you also remove this, please? This is standard Rails behaviour and unrelated to this feature, so it doesn't need to be tested. Sorry I missed this yesterday.

**Review by pablobm — COMMENTED** [review]
Just one tiny thing left!

File: spec/controllers/admin/customers_controller_spec.rb (hunk 2 above)
Comment (jubilee2) [reply spec/controllers/admin/customers_controller_spec.rb:73]: I want make sure action name is string not symbo, anyway I can remove it
```
      action =
        case action
        when "update" then "edit"
        when "create" then "new"
        else action
        end
```

**Review by pablobm — APPROVED** [review]

**Comment by pablobm** [comment]
Thank you again!

**Pull request merged**